clip.md

```diff
@@ -0,0 +1,443 @@
+
+   syntax ClipAuthLockStep
+   syntax ClipStep     ::= ClipLockStep
+   syntax ClipAuthStep ::= ClipAuthLockStep
+   syntax ClipLockStep ::= ClipAuthLockStep
+   syntax LockStep     ::= ClipContract "." ClipLockStep [klabel(clipStep)]
+   syntax LockAuthStep ::= ClipContract "." ClipAuthLockStep [klabel(clipStep)]
+// ---------------------------------------------
+   rule <k> lock Clip . _ => . ... </k>
+      <clip-locked> false => true </clip-locked>
+
+   rule <k> unlock Clip . _ => . ... </k>
+      <clip-locked> true => false </clip-locked>
+```
+
+Clip Data
+---------
+
+-   `ClipSale` tracks the parameters of an auction:
+
+    -   `pos`: Index in active array.
+    -   `tab`: Dai to raise, rad
+    -   `lot`: collateral to sell, wad
+    -   `usr`: Liquidated CDP
+    -   `tic`: Auction start time
+    -   `top`: Starting price, ray
+
+```k
+    syntax ClipSale ::= ClipSale ( pos: Int, tab: Rad, lot: Wad, usr: Address, tic: Int, top: Ray )
+ // -----------------------------------------------------------------------------------
+```
+
+Clip Events
+----------
+
+
+```k
+    syntax CustomEvent ::= Kick(id: Int, top: Ray, tab: Rad, lot: Wad, usr: Address, kpr: Address, coin: Wad) [klabel(Kick), symbol]
+ // --------------------------------------------------------------------------------------------------------------------------------
+
+    syntax ClipStep ::= "emitKick" Int Ray Rad Wad Address Address Wad
+ // ------------------------------------------------------------------
+    rule <k> emitKick ID TOP TAB LOT USR KPR COIN => ID ... </k>
+         <return-value> ID:Int </return-value>
+         <frame-events> ... (.List => ListItem(Kick(ID, TOP, TAB, LOT, USR, KPR, COIN))) </frame-events>
+
+    syntax CustomEvent ::= Redo(id: Int, top: Ray, tab: Rad, lot: Wad, usr: Address, kpr: Address, coin: Wad) [klabel(Redo), symbol]
+ //---------------------------------------------------------------------------------------------------------------------------------
+
+    syntax ClipStep ::= "emitRedo" Int Ray Rad Wad Address Address Wad
+ // --------------------------------------------------------
+    rule <k> emitRedo ID TOP TAB LOT USR KPR COIN => ID ... </k>
+         <return-value> ID:Int </return-value>
+         <frame-events> ... (.List => ListItem(Redo(ID, TOP, TAB, LOT, USR, KPR, COIN))) </frame-events>
+
+    syntax CustomEvent ::= Take(id: Int, max: Ray, price: Ray, owe: Wad, tab: Rad, lot: Wad, usr: Address) [klabel(Take), symbol]
+ // -----------------------------------------------------------------------------------------------------------------------------
+
+    syntax ClipStep ::= "emitTake" Int Ray Ray Wad Rad Wad Address
+ // --------------------------------------------------------------
+    rule <k> emitTake ID MAX PRICE OWE TAB LOT USR => ID ... </k>
+         <return-value> ID:Int </return-value>
+         <frame-events> ... (.List => ListItem(Take(ID, MAX, PRICE, OWE, TAB, LOT, USR))) </frame-events>
+```
+
+
+
+
+
+
```

Comment:
general nit: could cut down on whitespace in various places

Reply:
formatting errors fixed